### PR TITLE
Aliasing access happens in the runtime

### PR DESCRIPTION
**Pull request by andre4i**
# [358](https://dev.azure.com/fluidframework/internal/_workitems/edit/358/)

## Description

Effectively moving alias access from datastore to the runtime. While we're at it, make aliasing consistent with the id policy and disallow slashes (with the same type of error).

## PR Checklist

> Use the check-list below to ensure your branch is ready for PR. If the item is not applicable, leave it blank.

-   [ ] I have updated the documentation accordingly.
-   [x] I have added tests to cover my changes.
-   [x] All new and existing tests passed.
-   [x] My code follows the code style of this project.
-   [x] I ran the lint checks which produced no new errors nor warnings for my changes.
-   [x] I have checked to ensure there aren't other open Pull Requests for the same update/change.

## Does this introduce a breaking change?

-   [ ] Yes
-   [x] No

### CODE REVIEW

File: packages/runtime/container-runtime/src/containerRuntime.ts [hunk 1]
```diff
@@ -1925,7 +1931,8 @@ export class ContainerRuntime extends TypedEventEmitter<IContainerRuntimeEvents>
     }
 
     public async getRootDataStore(id: string, wait = true): Promise<IFluidRouter> {
-        const context = await this.dataStores.getDataStore(id, wait);
+        const internalId = this.maybeAlias(id);
+        const context = await this.dataStores.getDataStore(internalId, wait);
         assert(await context.isRoot(), 0x12b /* "did not get root data store" */);
```
Comment (agarwal-navin) [inline packages/runtime/container-runtime/src/containerRuntime.ts:1936]: This should not be an assert. This should be a usage error.

File: packages/runtime/container-runtime/src/dataStore.ts [hunk 2]
```diff
@@ -69,6 +70,10 @@ class DataStore implements IDataStore {
             default: unreachableCase(this.aliasState);
         }
 
+        if (alias.includes("/")) {
```
Comment (agarwal-navin) [inline packages/runtime/container-runtime/src/dataStore.ts:73]: This should probably be moved before the switch block.

File: packages/runtime/container-runtime/src/containerRuntime.ts [hunk 3]
```diff
@@ -1569,11 +1569,17 @@ export class ContainerRuntime extends TypedEventEmitter<IContainerRuntimeEvents>
         }
     }
 
+    private maybeAlias(id: string): string {
```
Comment (agarwal-navin) [inline packages/runtime/container-runtime/src/containerRuntime.ts:1572]: `getInternalId` might be better as it returns an id. The param can be called `maybeAlias`?

**Review by agarwal-navin — APPROVED** [review]

File: packages/runtime/container-runtime/src/dataStore.ts (hunk 2 above)
Comment (andre4i) [reply packages/runtime/container-runtime/src/dataStore.ts:73]: Was thinking about it and thought this is not relevant unless aliasing is actually happening.

File: packages/runtime/container-runtime/src/containerRuntime.ts (hunk 1 above)
Comment (andre4i) [reply packages/runtime/container-runtime/src/containerRuntime.ts:1936]: I would actually keep this as an assert. This should not happen. I mean other than keeping the id of root datastore, there's no id for regular datastores. So this method can only be called with garbage ids which would be acceptable for an assert imo

**Pull request merged**